internal/util/util.go

```diff
@@ -326,6 +326,18 @@ func GetVolumeSnapshotbackupWithStatusData(volumeSnapshotbackupNS string, volume
 			return false, errors.Wrapf(err, fmt.Sprintf("failed to get volumesnapshotbackup %s/%s", volumeSnapshotbackupNS, volumeSnapshotName))
 		}
 
+		if len(vsb.Status.Conditions) == 0 {
+			log.Infof("Waiting for volumesnapshotbackup %s to have conditions. Retrying in %ds", vsb.Name, interval/time.Second)
+			return false, nil
+		}
+
+		// check for status failure first
+		for _, condition := range vsb.Status.Conditions {
+			if condition.Status == metav1.ConditionFalse {
```

Comment:
@eemcmullan Should we check for the condition type as well ?

Reply:
@shubham-pampattiwar what about condition reason? It looks like condition type would be the same for failed and successful VSB: https://github.com/migtools/volume-snapshot-mover/blob/master/controllers/volumesnapshotbackup_controller.go#L126 

https://github.com/migtools/volume-snapshot-mover/blob/master/controllers/volumesnapshotbackup_controller.go#L137

Reply:
+1 on the reason check along with type.